chapter16/sxg/ex_16_2.pl

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+# copy STDOUT
+open( my $STDOUTOLD, '>&', STDOUT );
+
+# copy STDERR
+open( my $STDERROLD, '>&', STDERR );
+
+# open file handles
+open( STDOUT, '>>', 'ls.out' ) or die "Can't write to ls.out: $!";
+open( STDERR, '>>', 'ls.err' ) or die "Can't write to ls.err: $!";
+
+chdir '/' or die "Problem in chdir : $!";
+exec 'ls', '-l' or die "Problem in listing: $!";
+

```

Comment:
If you've used exec, none of the rest of the code is run. That might be why your file handles didn't seem to be reset.